application/cmdbabstract.class.inc.php

```diff
@@ -2665,10 +2720,11 @@ public function DisplayModifyForm(WebPage $oPage, $aExtraParams = array())
 		$sOwnershipToken = null;
 		$iKey = $this->GetKey();
 		$sClass = get_class($this);
-		$sMode = ($iKey > 0) ? static::ENUM_OBJECT_MODE_EDIT : static::ENUM_OBJECT_MODE_CREATE;
 
+		$this->SetDisplayMode(($iKey > 0) ? static::ENUM_OBJECT_MODE_EDIT : static::ENUM_OBJECT_MODE_CREATE);
```

Comment:
I know this feels a bit harsh, but feels more readable to me ? 

```suggestion
		if ($iKey > 0) {
				$this->SetDisplayMode(static::ENUM_OBJECT_MODE_EDIT);
		} else {
				 $this->SetDisplayMode(static::ENUM_OBJECT_MODE_CREATE);
		}
```